sdks/python/apache_beam/transforms/core.py

```diff
@@ -577,7 +672,7 @@ class ParDo(PTransformWithSideInputs):
   def __init__(self, fn_or_label, *args, **kwargs):
     super(ParDo, self).__init__(fn_or_label, *args, **kwargs)
 
-    if not isinstance(self.fn, DoFn):
+    if not (isinstance(self.fn, DoFn) or isinstance(self.fn, NewDoFn)):
```

Comment:
Done. As we'll remove the oldDoFn soon I don't think it might be worthwhile to create an abstract class